mass-merge.py

```diff
@@ -0,0 +1,203 @@
+#! /usr/bin/env python
+"""
+Merge signatures in bulk, based on an attribute in the spreadsheet.
+
+mass-merge.py takes a list of databases along with spreadsheets (w/-F),
+and merges signatures by values in the specified `merge-col`.
+
+The spreadsheet must contain two columns, 'ident' and the value of `--merge-col`;
+signatures are selected based on 'ident' and renamed to the value found in the merge column.
+Singletons (no additional signatures to merge) will be renamed for consistency.
+"""
+import sys
+import argparse
+import csv
+from collections import defaultdict
+
+import sourmash
+
+from sourmash.picklist import SignaturePicklist
+from sourmash.logging import set_quiet, error, notify, print_results, debug
+from sourmash import sourmash_args
+from sourmash.cli.utils import (add_moltype_args, add_ksize_arg)
+#from sourmash.sig import _check_abundance_compatibility
+
+def _check_abundance_compatibility(sig1, sig2):
+    if sig1.minhash.track_abundance != sig2.minhash.track_abundance:
+        raise ValueError("incompatible signatures: track_abundance is {} in first sig, {} in second".format(sig1.minhash.track_abundance, sig2.minhash.track_abundance))
+
+
+def massmerge(args):
+    """
+    rename one or more signatures.
+    """
+    set_quiet(args.quiet, args.quiet)
+    moltype = sourmash_args.calculate_moltype(args)
+    merge_col = args.merge_col
+
+    # load spreadsheets
+    merge_d = defaultdict(list)
+    all_idents=set()
+    for filename in args.from_spreadsheet:
+        count = 0
+        with open(filename, newline='') as fp:
+            first_entry= True
+            r = csv.DictReader(fp)
+            for row in r:
+                if first_entry:
+                    if not merge_col in r.fieldnames:
+                        error(f"ERROR on spreadsheet '{filename}'.")
+                        error(f"Merge column {merge_col} is not present.")
+                        sys.exit(-1)            
+                merge_name = row[merge_col]
+                ident = row['ident']
+                
+                assert ' ' not in ident, f"identifiers cannot have spaces - but '{ident}' does."
+                assert ident not in all_idents, f"duplicate identifer: '{ident}'"
+                all_idents.add(ident)
+
+                merge_d[merge_name].append(ident)
+                count += 1
+        notify(f"loaded {count} identifiers from '{filename}'")
+
+    num_merge_names = len(merge_d)
+    notify(f"found a total of {num_merge_names} distinct values for signature merging.")
+
+    # load each db and check that we can find all idents
+    found_idents = set()
+    idx_list = []
+    for db in args.dblist:
+        notify(f"loading index '{db}'")
+        idx = sourmash.load_file_as_index(db)
```

Comment:
you could do your ksize/moltype select right here, and then not repeat it again below (I think you do it two times total).

Reply:
`load_file_as_index` doesn't take `ksize` and `moltype`, right? But good point - I can do it when I do the `check_idx` to only do it once.

Reply:
that's correct, you can just do `idx = idx.select(...)` once.

Reply:
resolved via fc16963757a795163d3c17f24c00776012c87d23